From: Albert Astals Cid <aacid@kde.org>
Date: Tue, 2 Apr 2024 22:47:01 +0000 (+0200)
Subject: Use ifdef instead of if for __cpp_lib_span
X-Git-Url: https://codereview.qt-project.org/gitweb?p=qt%2Fqtbase.git;a=commitdiff_plain;h=a0439a7370df302e4a8f0402249b1aa2b1aeab4b;hp=710276ed85e8a5880ca73da149a94eb9dac49e32

Use ifdef instead of if for __cpp_lib_span

Like the other times it's used in this file

This is causing compilation errors in projects that use -Werror=undef

Fixes: QTBUG-123937
Pick-to: 6.7
Change-Id: I0cdd2910755dc9079890011dd8dbc27a6e64793e
---

diff --git a/src/corelib/tools/qspan.h b/src/corelib/tools/qspan.h
index c9de1005a778..d6ae2570ae3c 100644
--- a/src/corelib/tools/qspan.h
+++ b/src/corelib/tools/qspan.h
@@ -297,7 +297,7 @@ public:
         : QSpanBase(il.begin(), il.size())
     {}
 
-#if __cpp_lib_span
+#ifdef __cpp_lib_span
     template <typename S, size_t N, if_qualification_conversion<S> = true>
     Q_IMPLICIT constexpr QSpanBase(std::span<S, N> other) noexcept
         : QSpanBase(other.data(), other.size())
